python-checks/src/main/java/org/sonar/python/checks/UnusedLocalVariableCheck.java

```diff
@@ -120,17 +123,36 @@ private static boolean isVariableAccessedInStringTemplate(Symbol symbol, StringL
 
   public PreciseIssue createIssue(SubscriptionContext ctx, Symbol symbol, Usage usage) {
     if (isSequenceUnpacking(usage)) {
-      var quickFix = PythonQuickFix.newQuickFix(SEQUENCE_UNPACKING_QUICK_FIX_MESSAGE, TextEditUtils.replace(usage.tree(), "_"));
+      var quickFix = PythonQuickFix.newQuickFix(RENAME_QUICK_FIX_MESSAGE, TextEditUtils.replace(usage.tree(), "_"));
       var issue = ctx.addIssue(usage.tree(), String.format(SEQUENCE_UNPACKING_MESSAGE, symbol.name()));
       issue.addQuickFix(quickFix);
       return issue;
+    } else if (isLoopIndex(usage, symbol)) {
+      PreciseIssue issue = ctx.addIssue(usage.tree(), String.format(LOOP_INDEX_MESSAGE, symbol.name()));
+      PythonQuickFix quickFix = PythonQuickFix.newQuickFix(RENAME_QUICK_FIX_MESSAGE, TextEditUtils.replace(usage.tree(), "_"));
+      Symbol foundUnderscoreSymbol = null;
+      Tree searchTree = usage.kind().equals(Usage.Kind.LOOP_DECLARATION) ? ctx.syntaxNode() : null;
+      while (foundUnderscoreSymbol == null && searchTree != null) {
+        foundUnderscoreSymbol = ((FunctionDef) searchTree).localVariables().stream().filter(symbol1 -> "_".equals(symbol1.name())).findAny().orElse(null);
+        searchTree = TreeUtils.firstAncestorOfKind(searchTree, Kind.FUNCDEF);
+      }
```

Comment:
I'd suggest extracting this in a dedicated method whose purpose is to check whether the `_` symbol exists in another scope.